pubnub/crypto.py

```diff
@@ -103,3 +108,162 @@ def decrypt(self, key, file):
             result = unpad(cipher.decrypt(extracted_file), 16)
 
         return result
+
+
+class PubNubCryptoModule(PubNubCrypto):
+    FALLBACK_CRYPTOR_ID: str = '0000'
+    CRYPTOR_VERSION: str = 1
+    cryptor_map = {}
+    default_cryptor_id: str
+
+    def __init__(self, cryptor_map: Dict[str, PubNubCryptor], default_cryptor: PubNubCryptor):
+        self.cryptor_map = cryptor_map
+        self.default_cryptor_id = default_cryptor.CRYPTOR_ID
+
+    def register_cryptor(self, cryptor_id, cryptor_instance):
+        if len(cryptor_id) != 4:
+            raise PubNubException('Malformed cryptor_id.')
+
+        if cryptor_id in self.cryptor_map.keys():
+            raise PubNubException('Cryptor_id already in use')
+
+        if not isinstance(cryptor_instance, PubNubCrypto):
+            raise PubNubException('Invalid cryptor instance')
+
+        self.cryptor_map[cryptor_id] = cryptor_instance
+
+    def _validate_cryptor_id(self, cryptor_id: str) -> str:
+        cryptor_id = cryptor_id or self.default_cryptor_id
+
+        if len(cryptor_id) != 4:
+            logging.error(f'Malformed cryptor id: {cryptor_id}')
+            raise PubNubException('Malformed cryptor id')
+
+        if cryptor_id not in self.cryptor_map.keys():
+            logging.error(f'Unsupported cryptor: {cryptor_id}')
+            raise PubNubException('unknown cryptor error')
+        return cryptor_id
+
+    # encrypt string
+    def encrypt(self, message: str, cryptor_id: str = None) -> str:
+        cryptor_id = self._validate_cryptor_id(cryptor_id)
+        data = message.encode('utf-8')
+        crypto_payload = self.cryptor_map[cryptor_id].encrypt(data)
+        header = self.encode_header(cryptor_id=cryptor_id, cryptor_data=crypto_payload['cryptor_data'])
+        return b64encode(header + crypto_payload['data']).decode()
```

Comment:
What? Why? Why do we encode with base64 and then decode it? Or am I misreading it?

Reply:
`b64encode` produces binary string, and I want to have it as plain `string` so it's not b64 decoding more like cast from binary string to string